muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatRESTService.java

```diff
@@ -160,27 +168,44 @@ public Response chatStatus() {
       return Response.status(Status.FORBIDDEN).entity("Must be logged in").build();
     }
 
+    SchoolDataIdentifier identifier = sessionController.getLoggedUser();
+    
+    if (identifier == null) {
+        return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Couldn't find logged user").build();
+      }
+	  
     String enabledUsersCsv = pluginSettingsController.getPluginSetting("chat", "enabledUsers");
+    
     if (enabledUsersCsv == null) {
-      enabledUsersCsv = "";
+      return null;
     }
     List<String> enabledUsers = Arrays.asList(enabledUsersCsv.split(","));
-    SchoolDataIdentifier identifier = sessionController.getLoggedUser();
-    
-    if (identifier == null) {
-      return Response.status(Status.INTERNAL_SERVER_ERROR).entity("Couldn't find logged user").build();
+    boolean chatEnabled = false;
+
+    if (!enabledUsers.contains(identifier.toId())) {
+      return Response.ok(new StatusRESTModel(false, false, null)).build();
     }
     
+    UserChatSettings userChatSettings = chatController.findUserChatSettings(identifier);
+    
+    if (userChatSettings != null) {
+      UserChatVisibility visibility = userChatSettings.getVisibility();
+    	
+      if (visibility.toString().equals("VISIBLE_TO_ALL")) {
```

Comment:
`visibility == UserChatVisibility.VISIBLE_TO_ALL` should work and is better to compare two enums.